sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/encoders/EncoderResolutionSuite.scala

```diff
@@ -54,38 +54,38 @@ class EncoderResolutionSuite extends PlanTest {
     val encoder = ExpressionEncoder[StringLongClass]
 
     // int type can be up cast to long type
-    val attrs1 = Seq('a.string, 'b.int)
+    val attrs1 = Seq(attr("a").string, attr("b").int)
```

Comment:
We already have `"name".attr` (See the implicit `DslString`), can we extend it to support `"name".attr.string`?

Reply:
Actually, tt's not enough. `"name".attr` returns `UnresolvedAttribute` so we still fail to `"name".attr.map`.

Reply:
My general idea is, previous the DSL is based on symbol, can we update the DSL (those implicts) so that it's based on string? Ideally we should simply replace `'abc` to `"abc"` in the test code.

Reply:
It was my first idea but I thought it might be confusable that `"abc"` with implicit conversion doesn't look like an `attribute`. But DSL is for internal so it might be acceptable. What do you think?

Reply:
>  Ideally we should simply replace 'abc to "abc" in the test code.

Yes ideally it should be like what you suggested but I found it's difficult.
`DslString` implements `ImplicitOperators` but `DslSymbol` implements `ImplicitAttribute`. The implementation of `expr` conflicts. `DslString` implements `expr` as `Literal` but `DslSymbol` implements it as `UnresolvedExpression`.

So, the next option would be introduce a new syntax like `attr""` with `StringContext`. 

I said in the description like as follows.
> Another solution would be to introduce a new string interpolation syntax but it's difficult to have workaround when the syntax is conflict with another string interpolation syntax like that $"" defined in catalyst and core conflict so I don't adopt this solution.

But if this is internal usage only, it might be acceptable.